File: C++/Lab_10/Lab_10.cpp
// Lab 10
// Derek P Sifford
#include <iostream>
using namespace std;

int main() {

    char input;
    int group1 = 0, group2 = 0, sum = 0, i = 0;
    int n;

    cout << "Enter an integer\n";
    while (cin.get(input)) {
        if (input == '\n')
            break;

        n = int(input) - 48;

        if (i % 2 == 0) {
            group1 = group1 + n;
        }
        else {
            group2 = group2 + n;
        }
        i++;

    }

    if ((group1 + group2) % 2 == 0) {
        group1 = group1 * -1;
    } else {
        group2 = group2 * -1;
    }

    sum = group1 + group2;

    if (sum % 11 == 0) {
        cout << "The number is divisible by 11\n";
    } else {
        cout << "The number is not divisible by 11\n";
    }

}

File: golang/Lab_10/Lab_10.go
package main

import (
	"fmt"
)

func main() {

	var group1, group2 int
	var n []byte
	fmt.Scanln(&n)

	for i, num := range n {
		num -= 48

		if i%2 == 0 {
			group1 += int(num)
		} else {
			group2 += int(num)
		}
	}

	if check := group1 + group2; check%2 == 0 {
		group1 *= -1
	} else {
		group2 *= -1
	}

	if sum := group1 + group2; sum%11 == 0 {
		fmt.Println("The number is divisible by 11")
	} else {
		fmt.Println("The number is not divisible by 11")
	}

}
